### DISCUSSION TITLE
Can I ignore these warnings?

### DISCUSSION THREAD

**Discussion by mg-roku-shimizu** [pyinstaller/pyinstaller#8245] [Help] [answered]
I have got warnings below:

```
118 INFO: PyInstaller: 6.3.0
119 INFO: Python: 3.12.1
120 INFO: Platform: Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
121 INFO: wrote /**********.spec
19077 WARNING: Hidden import "packaging._elffile" not found!
19078 WARNING: Hidden import "packaging._manylinux" not found!
19078 WARNING: Hidden import "packaging._musllinux" not found!
19078 WARNING: Hidden import "packaging._parser" not found!
19078 WARNING: Hidden import "packaging._structures" not found!
19078 WARNING: Hidden import "packaging._tokenizer" not found!
19078 WARNING: Hidden import "packaging.markers" not found!
19078 WARNING: Hidden import "packaging.metadata" not found!
19078 WARNING: Hidden import "packaging.requirements" not found!
19078 WARNING: Hidden import "packaging.specifiers" not found!
19078 WARNING: Hidden import "packaging.tags" not found!
19078 WARNING: Hidden import "packaging.utils" not found!
19078 WARNING: Hidden import "packaging.version" not found!
19132 WARNING: Hidden import "jinja2" not found!
19285 WARNING: Library user32 required via ctypes not found
19372 WARNING: Library msvcrt required via ctypes not found
```

The build has succeeded, anyway.
Do I need take care of these, or can just ignore?

**Comment by bwoodsend** [comment, marked as answer]
Those ctypes libraries are triggered by some line of code like `ctypes.CDLL("msvcrt")` but since that's a Windows only library, it'll be inside an `if sys.platform == "windows"` block or something equivalent where they will be skipped at runtime but PyInstaller's dependency scanner is not smart enough to detect that their not ran on Linux. You can ignore those.

Hidden imports not being found can be an indicator that PyInstaller can't see your dependencies because your environment is a mess which is definitely a problem. If you don't use those packages though, you can ignore them.

**Reply by mg-roku-shimizu** [reply]
Thank you very much!
I understand that I can ignore these packages as I don't use them.

However, I am not sure if my environment is messy because I work in a virtual Python environment to keep things tidy as I have several versions of Python installed.
Is there any way that other versions or other software outside of the venv could interfere with PyInstaller?

**Reply by bwoodsend** [reply]
By messy I don't mean having packages you don't need. I mean people having their packages in a virtual environment but PyInstaller outside of it (usually PyCharm users since PyCharm guides naive users into using virtual environments without their knowing it or its implications), often then doing misguided workarounds like adding the virtual environment's `site-packages` directory to `sys.path` using PyInstaller's invariably misused `--paths` option.

**Reply by mg-roku-shimizu** [reply]
I see. My development environment doesn't fall into any of these categories (I use PyInstaller in venv and VSCode as an IDE), so I'll ignore these warnings for now.
I'm using Ubuntu 20.04, so Python 3.10>= versions are not available by default for apt commands, I'll investigate the possibility that I'm doing something wrong with the Python installation.
Thanks very much.
